test/test-apimatic-core/utilities/api_helper_test.rb

```diff
@@ -602,6 +602,37 @@ def test_deserialize_union_type
     ]
     assert_equal(expected, actual, 'Actual did not match the expected.')
   end
+
+  def test_get_additional_properties_success
```

Comment:
Add mocked models in the testing suite to have the integrated complex cases covered through them like simple, array, map of scalar & non-scalar types as additional properties, along with oneOf/anyOf cases as the type of additional properties.

Reply:
Also, the cases of the models with additional properties should be added in the JSON serialization and form encoding tests.

Reply:
I have added these case in the latest commit